Add container image signing to released images

### Proposal

Starting the new release we should support container image signing:

https://github.blog/2021-12-06-safeguard-container-signing-capability-actions/

https://github.com/kedacore/keda/blob/main/.github/workflows/release-build.yml

`cosing sign` fails:

https://github.com/kedacore/keda/runs/4901065794?check_suite_focus=true

```
Run make sign-images
COSIGN_EXPERIMENTAL=*** cosign sign -a GIT_HASH=ba37a***7a***686be9fa6d2cb***5c9a34***ecc***404e54 -a GIT_VERSION=main -a BUILD_DATE=2022.0***.2***.***9.46.45 ghcr.io/kedacore/keda:main
Generating ephemeral keys...
Retrieving signed certificate...
Non-interactive mode detected, using device flow.
Enter the verification code SGHM-QTLM in your browser at: https://oauth2.sigstore.dev/auth/device?user_code=SGHM-QTLM
Code will be valid for 300 seconds
Error: signing [ghcr.io/kedacore/keda:main]: getting signer: getting key from Fulcio: retrieving cert: error obtaining token: expired_token
main.go:46: error during command execution: signing [ghcr.io/kedacore/keda:main]: getting signer: getting key from Fulcio: retrieving cert: error obtaining token: expired_token
make: *** [sign-images] Error ***
Makefile:***97: recipe for target 'sign-images' failed
```

--- a/.github/workflows/main-build.yml
+++ b/.github/workflows/main-build.yml
@@ -8,6 +8,11 @@ jobs:
   validate:
     name: Validate
     runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+      id-token: write # needed for signing the images with GitHub OIDC Token **not production ready**
+
     # build-tools is built from ../../tools/build-tools.Dockerfile
     container: ghcr.io/kedacore/build-tools:main
     steps:

--- a/.github/workflows/release-build.yml
+++ b/.github/workflows/release-build.yml
@@ -7,6 +7,12 @@ jobs:
   build:
     name: Push Release
     runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+      deployments: write
+      id-token: write # needed for signing the images with GitHub OIDC Token **not production ready**
+
     # build-tools is built from ../../tools/build-tools.Dockerfile
     container: ghcr.io/kedacore/build-tools:main
     steps:

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -31,7 +31,7 @@
 
 - **General:** `keda-operator` Cluster Role: add `list` and `watch` access to service accounts ([#2406](https://github.com/kedacore/keda/pull/2406))|([#2410](https://github.com/kedacore/keda/pull/2410))
 - **General:** Delete the cache entry when a ScaledObject is deleted ([#2408](https://github.com/kedacore/keda/pull/2408))
-- **General:** Sign KEDA images published on GitHub Container Registry ([#2501](https://github.com/kedacore/keda/pull/2501))|([#2502](https://github.com/kedacore/keda/pull/2502))
+- **General:** Sign KEDA images published on GitHub Container Registry ([#2501](https://github.com/kedacore/keda/pull/2501))|([#2502](https://github.com/kedacore/keda/pull/2502))|([#2504](https://github.com/kedacore/keda/pull/2504))
 - **Azure Pipelines Scaler:** support `poolName` or `poolID` validation ([#2370](https://github.com/kedacore/keda/pull/2370))
 - **Graphite Scaler:** use the latest datapoint returned, not the earliest ([#2365](https://github.com/kedacore/keda/pull/2365))
 - **Kubernetes Workload Scaler:** ignore terminated pods ([#2384](https://github.com/kedacore/keda/pull/2384))

